### PR TITLE
CU-244p28y - Advanced open position features

### PR DESCRIPTION
**Pull request by 0xangelo**
## Issue
[CU-244p28y](https://app.clickup.com/t/244p28y)

## Description
Implements missing guard rails for the open_position extrinsics, as well as general improvements:
- fix(CH): check time before funding update at the end of close_position
- fix(CH): avoid underflow with saturating_sub
- chore(CH): update oracle TWAP in open_position
- chore(CH): settle with outstanding profits in open_position
- chore(CH): add index-mark spread controls to open_position
- test(CH): ensure no dust position left by open_position
- test(CH): ensure get_withdrawal_amounts helper does not panic
- refactor(CH): break down open_position in subfunctions
- refactor(CH): group trading helpers together
- style(CH): rename gains to profits
- docs(CH): add missing periods

_I recommend reviewers start with the test suite, as it outlines what the expected behaviors are._

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-244p28y Feature 4 (Open Position Functionality)](https://app.clickup.com/t/244p28y)

**Review by Cardosaum — APPROVED** [review]
Awesome work!

I don't have any comment to make in this one :sweat_smile: 

Great quality as always! :rocket:

File: frame/clearing-house/src/tests/mod.rs [hunk 1]
```diff
@@ -136,7 +136,7 @@ fn get_position(account_id: &AccountId, market_id: &MarketId) -> Option<Position
 }
 
 fn get_outstanding_gains(account_id: AccountId, market_id: &MarketId) -> Balance {
```
Comment (kevin-composable) [inline frame/clearing-house/src/tests/mod.rs:138]: should this function be change to `get_outstanding_profits` to match the change from `outstanding_gains` to `outstanding_profits`?

File: frame/clearing-house/src/lib.rs [hunk 2]
```diff
@@ -350,11 +353,11 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
-	/// Gains that were realized but cannot be withdrawn due to lack of offsetting realized losses
+	/// Profits that were realized but cannot be withdrawn due to lack of offsetting realized losses
```
Comment (kevin-composable) [inline frame/clearing-house/src/lib.rs:356]: add a .

Comment (kevin-composable) [reply frame/clearing-house/src/lib.rs:356]: I finally found a . that Matheus missed

**Review by kevin-composable — APPROVED** [review]

File: frame/clearing-house/src/tests/mod.rs (hunk 1 above)
Comment (0xangelo) [reply frame/clearing-house/src/tests/mod.rs:138]: Yeah that makes sense. I forgot to go through the test files to change the naming there too

File: frame/clearing-house/src/lib.rs (hunk 2 above)
Comment (0xangelo) [reply frame/clearing-house/src/lib.rs:356]: ahhahah, good catch!

Comment (Cardosaum) [reply frame/clearing-house/src/lib.rs:356]: hahaha

Good one Kevin! :joy:

**Pull request merged**